src/trex/molecule.py

```diff
@@ -1,19 +1,49 @@
 """
 A molecule is a DNA/RNA fragment that has potentially been sequenced multiple times
 """
+import math
 from typing import NamedTuple, List
-from collections import defaultdict
+from collections import defaultdict, Counter
 import numpy as np
 
 from .bam import Read
 
+# Threshold for low-complexity cloneIDs
+MIN_SHANNON_ENTROPY = 1.0
+
+
+def shannon_entropy(s: str) -> float:
+    """Return the empirical Shannon entropy in bits"""
+    n = len(s)
+    counts = Counter(s)
+    relative_frequencies = [count / n for count in counts.values()]
+    entropy = sum(-math.log2(freq) * freq for freq in relative_frequencies)
```

Comment:
Any reason for not using scipy?

Reply:
I hade the code already (written for a different project) and didn’t want to add the scipy dependency because it’s a 36 MB download. It didn’t feel justified for saving essentially two lines of straightforward code. However, I see that ship has already sailed because we depend on loompy, which in turn has scipy as a dependency, so I’ll change it.